### PR TITLE
Fix calloc returning junk data

### PR DESCRIPTION
**Pull request by hxef**
Thanks to Project579 and i860 for suggesting this change.

### CODE REVIEW

File: CommonLibF4/include/RE/Bethesda/MemoryManager.h [hunk 1]
```diff
@@ -289,7 +289,10 @@ namespace RE
 
 	[[nodiscard]] inline void* calloc(std::size_t a_num, std::size_t a_size)
 	{
-		return malloc(a_num * a_size);
+		auto ret = malloc(a_num * a_size);
+		if (ret)
+			std::memset(ret, 0, a_num * a_size);
```
Comment (Ryan-rsm-McKenzie) [inline CommonLibF4/include/RE/Bethesda/MemoryManager.h]:
```suggestion
		if (ret) {
			std::memset(ret, 0, a_num * a_size);
		}
```
Please always use braces for control structures

File: CommonLibF4/include/RE/Bethesda/BSExtraData.h [hunk 2]
```diff
@@ -619,11 +619,7 @@ namespace RE
 			return !((stl::to_underlying(a_type) - 11) & ~0x22u) && a_type != EXTRA_DATA_TYPE::kLeveledCreature;
 		}
 
-		void CreateFlags()
-		{
-			_flags = calloc<std::uint8_t>(N);
-			std::memset(_flags, 0, N);
-		}
+		void CreateFlags() { _flags = calloc<std::uint8_t>(N); }
```
Comment (Ryan-rsm-McKenzie) [inline CommonLibF4/include/RE/Bethesda/BSExtraData.h:622]: Can you convert this to a revert commit?

File: CommonLibF4/include/RE/Bethesda/MemoryManager.h [hunk 3]
```diff
@@ -289,7 +289,10 @@ namespace RE
 
 	[[nodiscard]] inline void* calloc(std::size_t a_num, std::size_t a_size)
 	{
-		return malloc(a_num * a_size);
+		auto ret = malloc(a_num * a_size);
```
Comment (Ryan-rsm-McKenzie) [inline CommonLibF4/include/RE/Bethesda/MemoryManager.h]:
```suggestion
		const auto ret = malloc(a_num * a_size);
```
Please always declare local variables `const` when appropriate

**Review by Ryan-rsm-McKenzie — CHANGES_REQUESTED** [review]

File: CommonLibF4/include/RE/Bethesda/BSExtraData.h (hunk 2 above)
Comment (hxef) [reply CommonLibF4/include/RE/Bethesda/BSExtraData.h:622]: Sure, this is the revert commit: https://github.com/Ryan-rsm-McKenzie/CommonLibF4/pull/10/commits/f7652b24f68516d204adacae6597e10091adbf6f

File: CommonLibF4/include/RE/Bethesda/MemoryManager.h (hunk 1 above)
Comment (hxef) [reply CommonLibF4/include/RE/Bethesda/MemoryManager.h]: Done.

File: CommonLibF4/include/RE/Bethesda/MemoryManager.h (hunk 3 above)
Comment (hxef) [reply CommonLibF4/include/RE/Bethesda/MemoryManager.h]: Done

**Comment by Ryan-rsm-McKenzie** [comment]
Thanks for the pr!

**Pull request merged**